roles/ceph-defaults/tasks/facts.yml

```diff
@@ -239,3 +239,30 @@
     - ceph_current_status['servicemap'] is defined
     - ceph_current_status['servicemap']['services'] is defined
     - ceph_current_status['servicemap']['services']['rgw'] is defined
+
+- name: get default value for osd_pool_default_pg_num
+  shell: |
+   {{ docker_exec_cmd }} ceph --cluster {{ cluster }} daemon mon.{{ hostvars[groups[mon_group_name][0]]['inventory_hostname'] }} config get osd_pool_default_pg_num
+  changed_when: false
+  failed_when: false
+  run_once: true
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  register: default_pool_default_pg_num
+
+- name: set_fact osd_pool_default_pg_num with default_pool_default_pg_num.stdout
+  set_fact:
+    osd_pool_default_pg_num: "{{ (default_pool_default_pg_num.stdout| from_json).osd_pool_default_pg_num | int }}"
+  run_once: true
+  delegate_to: "{{ item }}"
+  with_items: "{{ groups['all'] }}"
+  when: default_pool_default_pg_num.rc == 0
+
+- name: set_fact osd_pool_default_pg_num with pool_default_pg_num (backward compatibility)
```

Comment:
This task is here for backward compatibility for a while now.
I think it's time to add a deprecation note for this, @leseb wdyt?

Reply:
I think you're correct here, but I didn't see any harm in leaving it in with:
when: pool_default_pg_num is defined and pool_default_pg_num|length != 0